tests/tests.ts

```diff
@@ -210,6 +210,34 @@ describe('Tar stream build', () => {
 		})
 
 	})
+
+	it('should accept a pre-initialized dockerode object', function() {
+		this.timeout(60000)
+		return new Promise((resolve, reject) => {
+
+			const tarStream = fs.createReadStream('tests/test-files/archives/success.tar')
+
+			const hooks: BuildHooks = {
+				buildSuccess: () => {
+					resolve();
+				},
+				buildFailure: (e) => {
+					reject(e)
+				},
+				buildStream: (stream) => {
+					tarStream.pipe(stream)
```

Comment:
I'm not sure if the linter is ready yet, definitely need one though, I've been switching between ES6, coffeescript and typescript three times a day.